### PR TITLE
Bugfix: Ensure we have a valid hostname when adding certificates

### PR DESCRIPTION
**Pull request by jannfis**
This is a small bugfix to ensure that a valid hostname was given to `CreateCertificate` of API `CertificateService`, so that we don't end up with weird (and invalid) entries in the certificate list or in the worst case, with a security issue (although K8S API limits what can be used as a key in the ConfigMap already).

### CODE REVIEW

File: util/cert/cert.go [hunk 1]
```diff
@@ -63,6 +64,22 @@ const (
...
 
+// Regular expression that matches a valid hostname
+var validHostNameRegexp = regexp.MustCompile(`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9]))*$`)
+
+// Regular expression that matches a valid FQDN
+var validFQDNRegexp = regexp.MustCompile(`^([a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\.)+[a-zA-Z]{2,63}$`)
+
+// Can be used to test whether a given string represents a valid hostname
+// If fqdn is true, given string must also be a FQDN representation.
+func IsValidHostname(hostname string, fqdn bool) bool {
```
Comment (alexec) [inline util/cert/cert.go]: tests?

File: util/db/certificate.go [hunk 2]
```diff
@@ -169,6 +169,11 @@ func (db *db) CreateRepoCertificate(ctx context.Context, certificates *appsv1.Re
 	// Each request can contain multiple certificates of different types, so we
 	// make sure to handle each request accordingly.
 	for _, certificate := range certificates.Items {
+		// Ensure valid repo server name was given
+		if !certutil.IsValidHostname(certificate.ServerName, false) {
+			return nil, fmt.Errorf("Invalid hostname in request: %s", certificate.ServerName)
```
Comment (alexec) [inline util/db/certificate.go]: tests?

**Review by alexec — CHANGES_REQUESTED** [review]

File: util/cert/cert.go (hunk 1 above)
Comment (jannfis) [reply util/cert/cert.go]: Yup, commited

File: util/db/certificate.go (hunk 2 above)
Comment (jannfis) [reply util/db/certificate.go]: Yup, commited

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
needs linting

**Review by alexmt — APPROVED** [review]
LGTM. Please fix linting error

**Comment by jannfis** [comment]
Satisfied the almighty lint monster.

**Comment by alexec** [comment]
https://www.deviantart.com/khialat/art/Round-the-Twist-Lint-Monster-740912894

**Comment by jannfis** [comment]
> https://www.deviantart.com/khialat/art/Round-the-Twist-Lint-Monster-740912894

it looks as scary as it is!

**Pull request merged**